### PR TITLE
disabling jemalloc on bitrig

### PR DESCRIPTION
**Pull request by dhuseby**
it turns out that jemalloc doesn't behave well on bitrig.  with jemalloc enabled i get some kernel errors related to sbrk failures.  with jemalloc disabled, the errors go away.  i am investigating, but in the mean time, we should just disable jemalloc by default on bitrig.

### CODE REVIEW

**Comment by dhuseby** [comment]
While compiling Rust from source on Bitrig using a Rust snapshot built with jemalloc enabled, I get a lot of errors that look like this:

```
sbrk: grow 12850741248 failed, error = 12
sbrk: grow 8743018496 failed, error = 12
sbrk: grow 12560965632 failed, error = 12
sbrk: grow 12030480384 failed, error = 12
sbrk: grow 9737023488 failed, error = 12
sbrk: grow 11791466496 failed, error = 12
sbrk: grow 11408199680 failed, error = 12
sbrk: grow 10830114816 failed, error = 12
sbrk: grow 10409455616 failed, error = 12
sbrk: grow 9631027200 failed, error = 12
sbrk: grow 10049781760 failed, error = 12
```

I doesn't seem to affect Rust functionality but it sure looks scary.  Like I said, I'm disabling jemalloc on Bitrig until I know what it causing this and I upstream the patches to jemalloc.

**Comment by semarie** [comment]
@dhuseby I don't known if it is related, but under openbsd I use the following patch https://github.com/jasperla/openbsd-wip/blob/master/lang/rustc/patches/patch-src_jemalloc_configure for building jemalloc.

The latest version of jemalloc proper detect the lake of tls (thread-local-storage), but not the version used by rust.

I hope this could help you.

**Comment by dhuseby** [comment]
@semarie that might fix the problem.  i'll give it a try soon.  i did notice that without jemalloc, several of my test apps ran noticeably (wall clock) faster.  But that may have been because jemalloc was broken.

**Comment by dhuseby** [comment]
@nikomatsakis sorry for confusing things a bit, a new snapshot for bitrig was just uploaded and i added it to the snapshots.txt file and bumped llvm-auto-clean-trigger too.

**Comment by dhuseby** [comment]
@nikomatsakis both disabling jemalloc and the new snapshot are needed to get the bitrig buildbot working properly.

**Comment by alexcrichton** [comment]
You should be able to leave out the `llvm-auto-clean-trigger` modification as this PR didn't change LLVM itself (so it shouldn't need to be rebuilt)

**Comment by dhuseby** [comment]
@alexcrichton aha! ok, I see.

**Comment by dhuseby** [comment]
OK, I reverted the bump in llvm-auto-clean-trigger and rebased to the current master HEAD.

**Comment by alexcrichton** [comment]
Could you peel off the last two commits as well? Otherwise r=me, thanks!

**Comment by dhuseby** [comment]
@alexcrichton sure

**Comment by dhuseby** [comment]
@alexcrichton I'm confused about something with git.  I see you r+ with a link to a revision.  I updated my local checkout of master by pulling and I can see the commit when I run `git log c005f8a` but why isn't the commit showing up when I run `git log src/snapshots.txt`?  Is it because the commit is in another branch and not merged into master yet?  Is there an easy way to see what branch(es) a given revision in?

File: src/snapshots.txt [hunk 1]
```diff
@@ -1,4 +1,5 @@
 S 2015-03-07 270a677
+  birig-x86_64 4b2f11a96b1b5b3782d74bda707aca33bc179880
```
Comment (semarie) [inline src/snapshots.txt]: the name is invalid: it should be `bitrig` (missing `t`)

**Comment by dhuseby** [comment]
@semarie good catch! corrected.

**Comment by dhuseby** [comment]
rebased to current master HEAD.

**Pull request merged**